pkg/controller/machine/machine_controller_test.go

```diff
@@ -191,3 +192,90 @@ func TestReconcileRequest(t *testing.T) {
 		}
 	}
 }
+
+func TestReconcileFinalizers(t *testing.T) {
+	// If we need to add new finalizer logic later, add them here as well.
+	allFinalizers := []string{clusterv1.MachineFinalizer, metav1.FinalizerDeleteDependents}
+	nonNilClusterLen := len(allFinalizers)
+	nilClusterLen := nonNilClusterLen - 1
+
+	cluster1 := v1alpha1.Cluster{}
+
+	testCases := []struct {
+		cluster            *v1alpha1.Cluster
+		startingFinalizers []string
+		deleted            bool
+		expected           bool
+		expectedLen        int
```

Comment:
FinalizerDeleteDependents is dependent on a non-nil cluster. Do you think the tests would be stronger if we validated the contents of the reconciled finalizers, instead of just the length?

Reply:
@ncdc I mostly added the length check to protect against refactoring where machine is no longer passed by reference to ensure that the actual machine we want to update is being affected.  This is the most critical thing in this change, IMO.  Pretty much all the logic is identical.

I can add the explicit value check but that might make the matrix much more complex later.  In v1alpha2 it seems we're only adding one finalizer, so probably not a problem as we won't likely add more finalizers here.